xmtp/src/lib.rs

```diff
@@ -29,33 +33,31 @@ pub trait Errorer {
     type Error;
 }
 
-pub trait Store<I> {
-    fn store(&self, into: &mut I) -> Result<(), StorageError>;
+pub trait Store<DB> {
+    fn store(&self, into: &DB) -> Result<(), StorageError>;
 }
 
-pub trait KeyStore: Default + Errorer {
-    fn get_account(&mut self) -> Result<Option<Account>, StorageError>;
-
-    fn set_account(&mut self, account: &Account) -> Result<(), StorageError>;
+pub trait Fetch<T> {
+    fn fetch(&self) -> Result<Vec<T>, StorageError>;
 }
 
-#[derive(Debug, Error)]
-pub enum StorageError {
-    #[error("could not Fetch: {0}")]
-    Fetch(String),
-    #[error("could not Store: {0}")]
-    Store(String),
+pub trait Save<DB> {
+    fn save(&self, into: &DB) -> Result<(), StorageError>;
 }
 
-pub trait Fetch<T> {
-    fn fetch(&mut self) -> Result<Vec<T>, StorageError>;
+pub trait KeyStore: Default + Errorer {
+    fn get_account(&mut self) -> Result<Option<Account>, StorageError>;
+
+    fn set_account(&mut self, account: &Account) -> Result<(), StorageError>;
 }
 
 pub trait InboxOwner {
     fn get_address(&self) -> String;
     fn sign(&self, text: AssociationText) -> Result<RecoverableSignature, SignatureError>;
 }
 
+pub type PooledSqliteConnection = PooledConnection<ConnectionManager<SqliteConnection>>;
+
```

Comment:
Actually, I think that's leftover from a previous iteration of this where I was passing in the connection rather than the `EncryptedMessageStore` to things like `Store` and `Fetch`. Pretty sure I can delete.